UnitTestBot/UTBotJava
# Check if superclass or interfaces contain applicable types

## Description

Fixes #2450

## How to test

### Automated tests

Run `org.utbot.fuzzing.JavaTypesTest`

### Manual tests

1. Try run an example from a bug-report. It should work as expected.
2. Run modified example below. It should generate 3 tests: with null and other with only BImpl type:

```java
public class Implementations {

    public interface A<T> {
        T getValue();
    }

    public static class AImpl implements A<String> {

        private final String value;

        public AImpl(String value) {
            this.value = value;
        }

        @Override
        public String getValue() {
            return value;
        }
    }

    public static class BImpl implements A<Integer> {

        private final Integer value;

        public BImpl(Integer value) {
            this.value = value;
        }

        @Override
        public Integer getValue() {
            return value;
        }
    }

    public static int test(A<Integer> value) {
        if (value.getValue() < 0) {
            return 0;
        }
        return value.getValue();
    }
}
```

## Self-check list
